### PR TITLE
bump to 4.3 to fix #244

### LINKED ISSUE #244: olm data needs to be update for 4.3
https://github.com/openshift/cluster-logging-operator/tree/release-4.3/manifests
file not found: /manifests/4.3/image-references

### PR DESCRIPTION
**Pull request by jcantrill**
fix #244 by bumping to 4.3

### CODE REVIEW

**Comment by jcantrill** [comment]
cc @ewolinetz

File: manifests/4.3/cluster-logging.v4.3.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ metadata:
     createdAt: 2018-08-01T08:00:00Z
     support: AOS Logging
     # The version value is substituted by the ART pipeline
-    olm.skipRange: ">=4.1.0 <4.2.0"
+    olm.skipRange: ">=4.2.0 <4.3.0"
```
Comment (ewolinetz) [inline manifests/4.3/cluster-logging.v4.3.0.clusterserviceversion.yaml]: same question here as EO... should we go from `>=4.1.0 <4.3.0`?

Comment (jcantrill) [reply manifests/4.3/cluster-logging.v4.3.0.clusterserviceversion.yaml]: No.  This is intentional and fits our policy with prior releases to force stepwise upgrades

**Comment by bparees** [comment]
@jcantrill @richm this has failed a bunch of times on:

```
FAILURE after 119.732s: hack/testing/test-020-olm-upgrade.sh:93: executing 'oc -n openshift-logging get deployment cluster-logging-operator -o jsonpath={.spec.template.spec.containers[0].image}' expecting any result and text 'registry.svc.ci.openshift.org/ci-op-pthzmzl7/stable:cluster-logging-operator'; re-trying every 0.2s until completion or 120.000s: the command timed out
```

seems like some investigation is needed.

**Comment by bparees** [comment]
this is the output you're actually getting:
```
registry.redhat.io/openshift4/ose-cluster-logging-operator:v4.1.18-201909211522
```

so something isn't substituting properly in your tests.

**Comment by jcantrill** [comment]
/hold cancel

update skip range once 4.2 is released https://github.com/openshift/cluster-logging-operator/issues/248

File: manifests/4.3/cluster-logging.v4.3.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ metadata:
     createdAt: 2018-08-01T08:00:00Z
     support: AOS Logging
     # The version value is substituted by the ART pipeline
-    olm.skipRange: ">=4.1.0 <4.2.0"
+    olm.skipRange: ">=4.1.0 <4.3.0"
```
Comment (bparees) [inline manifests/4.3/cluster-logging.v4.3.0.clusterserviceversion.yaml]: i really meant your test code should hack this, not that you should check it in this way, but i guess it's fine as long as you change it before 4.3 ships.  (or you're ok w/ users upgrading from 4.1 directly to 4.3)

Comment (jcantrill) [reply manifests/4.3/cluster-logging.v4.3.0.clusterserviceversion.yaml]: This is why I logged #248.  I don't think we should but until we can get a cluster that has OLM content that is appropriate for a given release we'll have to do this

**Comment by richm** [comment]
@jcantrill I believe this is the problem
```
hack/test-e2e.sh
[INFO] running e2e hack/testing/test-001-e2e-legacy.sh 
+ NAMESPACE=openshift-logging
+ os::test::junit::declare_suite_start hack/testing/test-001-e2e-legacy.sh
+ local suite_name=hack/testing/test-001-e2e-legacy.sh
+ local num_suites=0
+ echo '=== BEGIN TEST SUITE github.com/openshift/origin/test/hack/testing/test-001-e2e-legacy.sh ==='
+ NUM_OS_JUNIT_SUITES_IN_FLIGHT=1
+ export NUM_OS_JUNIT_SUITES_IN_FLIGHT
+ ARTIFACT_DIR=/tmp/artifacts
+ VERSION=1.10.8
...
+ CSV_FILE=/go/src/github.com/openshift/cluster-logging-operator/manifests/1.10.8
+ EO_CSV_FILE=/go/src/github.com/openshift/cluster-logging-operator/vendor/github.com/openshift/elasticsearch-operator/manifests/1.10.8
+ trap cleanup exit
+ '[' -n '' ']'
+ '[' -n 'registry.svc.ci.openshift.org/ci-op-my8f7wtk/stable:${component}' ']'
++ sed -e 's,${component},cluster-logging-operator,' /dev/fd/63
+++ echo 'registry.svc.ci.openshift.org/ci-op-my8f7wtk/stable:${component}'
+ IMAGE_CLUSTER_LOGGING_OPERATOR=registry.svc.ci.openshift.org/ci-op-my8f7wtk/stable:cluster-logging-operator
+ KUBECONFIG=/tmp/admin.kubeconfig
+ /go/src/github.com/openshift/cluster-logging-operator/hack/gen-olm-artifacts.sh /go/src/github.com/openshift/cluster-logging-operator/manifests/1.10.8 openshift-logging ns
+ oc create -f -
Traceback (most recent call last):
  File "/go/src/github.com/openshift/cluster-logging-operator/hack/gen-olm-artifacts.sh", line 164, in <module>
    csv = loadFile(csvFile)
  File "/go/src/github.com/openshift/cluster-logging-operator/hack/gen-olm-artifacts.sh", line 14, in loadFile
    with open(file, "r") as stream:
IOError: [Errno 2] No such file or directory: '/go/src/github.com/openshift/cluster-logging-operator/manifests/1.10.8'
error: no objects passed to create
```

It seems that `VERSION` is being set somewhere in the environment - maybe it is the go version? - but it is overriding the expected ocp version.  I think you either need to change Makefile to unconditionally set `VERSION` - instead of
`export VERSION?=4.3`
use
`export VERSION=4.3`
or, if you need to be able to override `VERSION`, you'll have to use a different env var name e.g. `OCP_VERSION` or perhaps there is already one used for this purpose in CI or in some other testing code.

**Comment by jcantrill** [comment]
> or, if you need to be able to override `VERSION`, you'll have to use a different env var name e.g. `OCP_VERSION` or perhaps there is already one used for this purpose in CI or in some other testing code.

modified to `OCP_VERSION`

File: hack/testing/test-010-deploy-via-olm-minimal.sh [hunk 3]
```diff
@@ -42,7 +42,7 @@ fi
 if [ -n "${IMAGE_FORMAT:-}" ] ; then
   IMAGE_CLUSTER_LOGGING_OPERATOR=$(sed -e "s,\${component},cluster-logging-operator," <(echo $IMAGE_FORMAT))
 else
-  IMAGE_CLUSTER_LOGGING_OPERATOR=${IMAGE_CLUSTER_LOGGING_OPERATOR:-registry.svc.ci.openshift.org/origin/4.2:cluster-logging-operator}
+  IMAGE_CLUSTER_LOGGING_OPERATOR=${IMAGE_CLUSTER_LOGGING_OPERATOR:-registry.svc.ci.openshift.org/origin/${VERSION}:cluster-logging-operator}
```
Comment (richm) [inline hack/testing/test-010-deploy-via-olm-minimal.sh]: OCP_VERSION

File: hack/testing/test-020-olm-upgrade.sh [hunk 4]
```diff
@@ -60,12 +60,12 @@ fi
 if [ -n "${IMAGE_FORMAT:-}" ] ; then
   IMAGE_CLUSTER_LOGGING_OPERATOR=$(sed -e "s,\${component},cluster-logging-operator," <(echo $IMAGE_FORMAT))
 else
-  IMAGE_CLUSTER_LOGGING_OPERATOR=${IMAGE_CLUSTER_LOGGING_OPERATOR:-registry.svc.ci.openshift.org/origin/4.2:cluster-logging-operator}
+  IMAGE_CLUSTER_LOGGING_OPERATOR=${IMAGE_CLUSTER_LOGGING_OPERATOR:-registry.svc.ci.openshift.org/origin/4.3:cluster-logging-operator}
```
Comment (richm) [inline hack/testing/test-020-olm-upgrade.sh]: can you use the env. var. OCP_VERSION here?

**Pull request merged**